examples/platforms/nrf52840/alarm.c

```diff
@@ -172,46 +183,51 @@ static inline otPlatUsecAlarmTime AlarmGetCurrentTime(void)
         offset = sTimeOffset;
     }
 
-    return TimeAdd(offset, TicksToTime(rtcValue2));
-}
+    if (aIndex == kMsTimer)
+    {
+        currentTime = TimeAdd(offset, TicksToTime(rtcValue2, aIndex));
+    }
+    else
+    {
+        currentTime = TimeAdd(offset * US_PER_MS, TicksToTime(rtcValue2, aIndex));
+    }
 
-static inline otPlatUsecAlarmTime AlarmGetCurrentTimeRtcProtected(void)
-{
-    return TimeAdd(AlarmGetCurrentTime(), (otPlatUsecAlarmTime) {0, 2 * US_PER_TICK});
+    return currentTime;
 }
 
-static inline bool TimeIsLower(otPlatUsecAlarmTime aTime1, otPlatUsecAlarmTime aTime2)
+static inline uint32_t AlarmGetCurrentTimeRtcProtected(AlarmIndex aIndex)
 {
-    return ((aTime1.mMs < aTime2.mMs) ||
-            ((aTime1.mMs == aTime2.mMs) && (aTime1.mUs < aTime2.mUs)));
-}
+    uint32_t currentTime;
 
-static inline bool AlarmShallStrike(otPlatUsecAlarmTime aNow, AlarmIndex aIndex)
-{
-    if (TimeIsLower(sTimerData[aIndex].mTargetTime, sTimerData[aIndex].mT0))
+    if (aIndex == kMsTimer)
     {
-        // Handle situation when timespan included timer overflow.
-        if (TimeIsLower(aNow, sTimerData[aIndex].mT0) && !TimeIsLower(aNow, sTimerData[aIndex].mTargetTime))
-        {
-            return true;
-        }
+        currentTime = AlarmGetCurrentTime(aIndex);
```

Comment:
I'm not sure if it will work correctly without adding 2 ticks. @rlubos What do you think?